File: zad_1/hash.rb


tablica = []

def haszowanie(tablica, klu)
	indeks =0
	sukces =0
	indeks = klu%103
	
	while sukces < 1   do
		if tablica[indeks] = 0
			puts indeks
			tablica[indeks] = klu
			sukces =1
		else
			indeks++
			if indeks >= 103
				indeks=0
			end
		end
	end
end

for i in 1..100
 tablica << 0
end

puts "Podaj n: "
n=gets.to_i

for i in 1..10
	puts "podaj element nr: " + i.to_s
	
	klucz = gets.to_i
	
	haszowanie tablica, klucz
end

for j in 1..103
	puts tablica[j]
end

File: zad_3/haszowanie/hash.cpp
    // haszowanie liniowe

    #include <iostream>

    using namespace std;

    void haszowanie(int tablica[], int klucz);

    int main()
    {
      int tablica[103]={0},n,klucz;
      cout << "Podaj n: ";
      cin >> n;
      for(int i=0; i<n; i++)
        {
          cout << "Podaj element nr: " << i+1 << ": ";
          cin >> klucz;
          haszowanie(tablica, klucz); // umieszczamy klucz w wyhaszowanym indeksie tablicy
        }
      for(int j=0; j<103; j++) // po hashowaniu, wypisujemy tablice
        {
          cout << tablica[j] << ", ";
        }
      return 0;
    }

    void haszowanie(int tablica[], int klucz)
    {
      int indeks,sukces=0;
      indeks=klucz%103;  // wyliczenie indeksu
      while(!sukces)  // dopoki nie sukces
        {
          if(!tablica[indeks]) // jezeli nie ma kolizji, tzn tablica[indeks]=0
          {
             tablica[indeks]=klucz; // wpisanie do tablicy
             sukces=1;  // wolne, dalej sie while nie wykona w tym wywolanie funkcji
          }
          else // kolizja!
          {
             indeks++; // szukamy nastepnego wolnego miejsca
             if(indeks>=103) indeks=0; // jak dojedzie do konca tablicy to zerujemy indeks
          }
        }
    }


